tests/checkpointing/unit/test_async_writer.py

```
@@ -125,41 +140,59 @@ def test_async_is_equivalent_to_sync(self, tmp_path_dist_ckpt):
                     loaded_sync_state_dict[key], state_dict[key]
                 ), f"Mismatch for key '{key}' between async checkpoint and original state_dict."
 
-    def test_cached_metadata(self, tmp_path_dist_ckpt):
+    def test_errors_are_reported(self, tmp_path_dist_ckpt, async_queue):
```

Is async_queue setup and teardown needed between tests?
The point is that even our training loop may get an exception from CP error and may choose to ignore and continue. In that case async_queue need not be setup again.

---

I can change it to a session-scoped fixture. However that reduces test isolation. If we want to test the above case (ignore exception, keep using the async loop) I think we should make that an explicit test case. I can add it to the list if you agree.